EIPS/eip-4902.md

```diff
@@ -0,0 +1,131 @@
+
+- roleToTrustConfig, map(uint8 => TrustConfig)
+Defines the required proofs according to specific role.
+So for example, role of guest will require trust of first pay of 5$ and deposit of 50$ and role of friend will required 0$ of both.
+
+
+Entity can interact with DAA with following API
+
+    /**
+     * Proves that 'account' has the requried proofs in order to connect with the access.
+     * */
+    function prove(
+        uint256 accessId,
+        address account,
+        uint8 role
+    ) external returns (bool);
+
+    /**
+     * Creates the connection for given 'accessId' , 'account' and 'role'.
+     * It depends on the role whether it allows it or not, based on the proofs and authorization configuration.
+     * */
+    function open(
+        uint256 accessId,
+        address account,
+        uint8 role,
+        uint256 startTime
+    ) external returns (bool);
+
+    /**
+     * Initating the trust process between the actors.
+     *
+     * Trust can be build by:
+     * 1. Transferring the money at once (Buy something, or get one time access. - example: ticket to party)
+     * 2. Trasnfering the money with inital amount and final amount when finish , while the money in deposit
+     * 3. Transferring the money in a stream way, with fixed rate and money. which represent the amount of job that has been done - example: taxi ride
+     * 4. Transferring the money in stages. for example professional service, that split his job into smaller ones and charge for each one
+     * 5. 3rd party trust - Transfer the funds as a deposit to the operator, who verifies that everything is in order and, if there is a problem, decides where the funds should be transferred.
+     * */
+    function trust(uint256 accessId, address account) external returns (bool);
+
+    /**
+    * Closing the connection for 'account' with the DAA
+    * */
+    function close(uint256 accessId, address account) external returns (bool);
+
+## Rationale
+Its the very first draft.
+There is consideration to split this into proof, trust , access instead of being in one contract.
+For me its more a philosfical considoration that im wondering if an access can be represented without this things based on "An access emerges from the abilities of actors to benefit from ‘things’."
+I hope people will help me to achieve that conclusions in the future. 
+
+## Backwards Compatibility
+There is some desire to made it possible to take existing ERC-721 tokens and made them an access too.
+
+## Test Cases
+Test cases for an implementation are mandatory for EIPs that are affecting consensus changes.  If the test suite is too large to reasonably be included inline, then consider adding it as one or more files in `../assets/eip-####/`.
+
+## Reference Implementation
+In private repo for now, so will be open
+
+## Security Considerations
+Every action should be accepted by the actors.
+
+## Copyright
+Eartho Group, Daniel, Tel Aviv
```

Comment:
The copyright must be CC0 as specified in EIP-1. Please review EIP-1 for the exact text.